app/src/main/java/com/example/plantsapp/presentation/ui/tasks/TasksViewModel.kt

```diff
@@ -43,8 +44,20 @@ class TasksViewModel @AssistedInject constructor(
         _plantsWithTasks.value =
             plantsRepository.fetchPlants()
                 .map { plant ->
-                    plant to getTasksForPlantAndDateUseCase(plant, date)
+                    plant to
+                            getTasksForPlantAndDateUseCase(plant, date)
+                                .mapToTaskWithState()
                 }
                 .filter { it.second.isNotEmpty() }
     }
+
+    private fun List<Pair<Task, Boolean>>.mapToTaskWithState(): List<TaskWithState> {
```

Comment:
How about change this to work only with pair, not with list? 
And use map outside?